npm/ng-packs/packages/core/src/lib/abstracts/auth.service.ts

```diff
@@ -56,4 +65,6 @@ export interface IAuthService {
   navigateToLogin(queryParams?: Params): void;
 
   login(params: LoginParams): Observable<any>;
+
+  loginUsingGrant(grantType: string, parameters: object, headers?: HttpHeaders): Promise<AbstractModel>;
```

Comment:
this method can be optional

Reply:
I think we can't mark as optional. That used on Abp commercial. I know the user have not use Oauth but we need that function for impersonate and link account.